packages/dataviews/src/components/dataviews-filters/search-widget.tsx

```diff
@@ -84,22 +86,37 @@ const getNewValue = (
 	return [ value ];
 };
 
+function generateCompositeItemId(
+	prefix: string,
+	filterElement: NormalizedFilter[ 'elements' ][ number ]
+) {
+	return `${ prefix }-${ filterElement.value }`;
+}
+
 function ListBox( { view, filter, onChangeView }: SearchWidgetProps ) {
-	const compositeStore = useCompositeStore( {
-		virtualFocus: true,
-		focusLoop: true,
-		// When we have no or just one operator, we can set the first item as active.
-		// We do that by passing `undefined` to `defaultActiveId`. Otherwise, we set it to `null`,
-		// so the first item is not selected, since the focus is on the operators control.
-		defaultActiveId: filter.operators?.length === 1 ? undefined : null,
-	} );
+	const baseId = useInstanceId( ListBox, 'dataviews-filter-list-box' );
+
+	const [ activeCompositeId, setActiveCompositeId ] = useState<
+		string | null | undefined
+	>(
+		// When there are one or less operators, the first item is set as active
+		// (by setting the initial `activeId` to `undefined`).
+		// With 2 or more operators, the focus is moved on the operators control
+		// (by setting the initial `activeId` to `null`), meaning that there won't
+		// be an active item initially. Focus is then managed via the
+		// `onFocusVisible` callback.
+		filter.operators?.length === 1 ? undefined : null
+	);
 	const currentFilter = view.filters?.find(
 		( f ) => f.field === filter.field
 	);
 	const currentValue = getCurrentValue( filter, currentFilter );
 	return (
 		<Composite
-			store={ compositeStore }
+			virtualFocus
+			focusLoop
+			activeId={ activeCompositeId }
```

Comment:
We are using to `activeId` instead of `defaultActiveId` since we are switching from using `Composite` uncontrolled, to controlling its active item ID state.

I moved the `defaultActiveId` value to the initial state value for `activeCompositeId`.

I'll look more into it, and report any findings

Reply:
This logic affects the composite widget when opening the dropdown — it basically prevents `Composite` from automatically picking an active composite item if there is more than one filter operator.

I reviewed the code, and I don't think this change will introduce differences in the current state of the code since the `filter.operators` array doesn't seem to change while the dialog is open. 

If this assumption was ever going to change, I suspect the previous version would have also incurred in some unexpected behaviour

Reply:
In general, the current implementation feels a bit hacky. I wish we didn't have to control the `activeId` state at all. With more and and a willingness to tweak this UI a bit, we could also consider having different styles for active vs focused items.

But not in the scope for this PR, and without a high priority.